### PR TITLE
add pdf artifact upload step

### PR DESCRIPTION
**Pull request by max-ostapenko**
- [x] generated PDFs are temporarily stored in Action Artifacts
- [x] deployment script uses artifact ZIP as a source for deployed PDFs
- [x] PDF files are ignored in the repo moving forward

### CODE REVIEW

**Comment by tunetheweb** [comment]
How's this looking @max-ostapenko ? About to do a release, but will hold off for a day or two if you think this is close to being completed and wanna test it out on a release?

Still not sure how you can download these assets locally?

**Comment by max-ostapenko** [comment]
@tunetheweb in progress, checking both artifacts and LFS.
Will adapt `src/tools/scripts/deploy.sh` and then you can test it out for your release.

Any other scripts impacted?

**Comment by tunetheweb** [comment]
Cool will hold off on release then.

I think it should just be that script and the action which generates the PDF.

File: src/tools/scripts/deploy.sh [hunk 1]
```diff
@@ -98,10 +98,30 @@ if [ "$(pgrep -f 'python main.py')" ]; then
...
+
+# Download latest e-books from Github Action artifacts
+echo "Downloading latest e-books from Github Action artifacts"
+# Set your GitHub repository and workflow name
+REPO_OWNER=HTTPArchive
+REPO_NAME=almanac.httparchive.org
+WORKFLOW_ID=predeploy.yml
+# Set your Github personal access token
+export $(cat ../.env | xargs)
+#read -sp 'Enter your Github personal access token: ' GITHUB_TOKEN
```
Comment (max-ostapenko) [inline src/tools/scripts/deploy.sh:115]: @tunetheweb please adjust to whatever is preferred option to pass GITHUB_TOKEN value.
Here I tested with `.env` file and user input.

File: src/tools/scripts/deploy.sh [hunk 2]
```diff
@@ -98,10 +98,31 @@ if [ "$(pgrep -f 'python main.py')" ]; then
...
+# Download latest e-books from Github Action artifacts
+echo "Downloading latest e-books from Github Action artifacts"
+# Set your GitHub repository and workflow name
+REPO_OWNER=HTTPArchive
+REPO_NAME=almanac.httparchive.org
+WORKFLOW_ID=predeploy.yml
+# Set your Github personal access token
+# shellcheck disable=SC1091
+source .env
+#read -sp 'Enter your Github personal access token: ' GITHUB_TOKEN
```
Comment (tunetheweb) [inline src/tools/scripts/deploy.sh:116]: Let's just get deployers to set a GITHUB_TOKEN env variable themselves and remove these lines.

Can we add this to here some where? https://github.com/HTTPArchive/almanac.httparchive.org/blob/bed8e5eab20f28633dd2040562379023b7628723/src/README.md?plain=1#L321

Should also explain how to generate an access token and the permission it requires (looks to me `public_repo`).

Also might need to update these steps: https://github.com/HTTPArchive/almanac.httparchive.org/blob/bed8e5eab20f28633dd2040562379023b7628723/src/README.md?plain=1#L359-L372

**Review by tunetheweb — CHANGES_REQUESTED** [review]
This is a VERY nice solution to the problem. Thanks SO much for figuring this out.

Just a few small comments and then think good to merge.

File: src/README.md [hunk 3]
```diff
@@ -360,8 +368,9 @@ The deploy script will do the following:
 - Ask you to confirm you've run the pre-deploy script via GitHub Actions
 - Switch to the production branch
 - Merge changes from main
-- Do a clean install
-- Run the tests
+- Do a clean install (remove generated chapters and e-books)
+- Downloade latest e-books from Github Action artifacts
```
Comment (tunetheweb) [inline src/README.md:372]:
```suggestion
- Download latest e-books from Github Action artifacts
```

File: src/README.md [hunk 4]
```diff
@@ -332,6 +332,14 @@ You must first do some setup locally:
 gcloud init
 ```
 
+3. Set Github personal access token to GITHUB_TOKEN environment variable
+
+Use existing or create a new token [in GitHub.com](https://github.com/settings/personal-access-tokens/new). In token configuration select *Repository access = Public Repositories (read-only)*.
```
Comment (tunetheweb) [inline src/README.md:337]:
```suggestion
Use an existing token or create a new token [in GitHub.com](https://github.com/settings/personal-access-tokens/new). In token configuration select *Repository access = Public Repositories (read-only)*.
```

File: src/README.md [hunk 5]
```diff
@@ -332,6 +332,14 @@ You must first do some setup locally:
 gcloud init
 ```
 
+3. Set Github personal access token to GITHUB_TOKEN environment variable
+
+Use an existing token or create a new token [in GitHub.com](https://github.com/settings/personal-access-tokens/new). In token configuration select *Repository access = Public Repositories (read-only)*.
+
+```bash
+export GITHUB_TOKEN=github_pat_...
```
Comment (tunetheweb) [inline src/README.md:340]: My token begins with `ghp_` not `github_pat_`:

```suggestion
export GITHUB_TOKEN=...
```

**Comment by tunetheweb** [comment]
Thanks again for this @max-ostapenko . I'm actually travelling for the next few days so will pick this up after that if you don't mind. But looks good to merge to me.

**Comment by max-ostapenko** [comment]
Enjoy the travel, and let me know if I can assist further.

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**